content/docs/minion-v0.4/tag_cache.md

```diff
@@ -37,21 +35,22 @@ However, we can increase it further.
 
 ## General concept of a hierarchical tag cache
 
-In this release, we implemented an optimised hierarchical tag cache to
-compress the empty (unset) tags cached in the tag cache.
-This compression is based on the observation that most of the memory is not tagged for most use cases.
-It is likely that most of the tags cached are zero or empty in normal operation condition.
-To utilise this pattern, a hierarchical cache uses a bit-map to record the emptiness of each cached tag cache line.
-When a whole line is empty (0), the emptiness state stored in the bit-map is enough to fully represent the cache line;
-therefore, the actual empty line is no long need to be cached (no need to access it from memory as well).
+In this release, we observe that most of the memory is not tagged for most use cases.
+Therefore, we can implement optimisation based on an hierarchical tag cache to
+compress the unset tags cached in the tag cache.
+It is likely that most of the tags cached are unset in normal operation condition.
+By analogy with the cache lines of recently used code, tag cache lines contain recently used tags.
+A tag cache line is just a normal cache line in the tag cache; however, this cache line stores tags rather than data.
+To utilise this pattern, a hierarchical cache uses a bit-map to record the presence of any cached tag cache line.
+When a whole line is unused, the bit-map is enough to fully represent the cache line;
+therefore, the actual unused line is no long needed to be cached (no need to access it from memory as well).
```

Comment:
We cannot really say it is unused. A tag of 0 may be the actual value needed by a program. In this case, the cache line is used, but there is no needed to store it in the tag cache.